groups/Pauli.group

```diff
@@ -3,14 +3,14 @@
     <gapid>16,13</gapid>
     <gapname>(C4 x C2) : C2</gapname>
     <link>http://groupnames.org/1/C4oD4.html</link>
```

Comment:
This links to https://people.maths.bris.ac.uk/~matyd/GroupNames/1/C4oD4.html, but I noticed that the presentation is different. Group Names shows: 

G = < a,b,c | a^4=c^2=1, b^2=a^2, ab=ba, ac=ca, cbc=a^2b >

whereas GE gives: 

<a, b, c: a^4=b^2=c^2=1, cbc=ba^2, ab=ba, ac=ca> 

which differs since a^2 != 1 but GE has b^2 = 1 while Group Names has b^2 = a^2. In both cases this is supposed to correspond with SmallGroup(16, 13). Are these really equivalent?

Reply:
Hi, Jon.

Thanks for diving into the code and helping improve the software!

It looks like you have a commit here that you're confident is correct, then a question that you're less confident about and we should try to investigate together.  Am I understanding that correctly?  I just want to make sure that I follow up in the right way.

Nathan

Reply:
Hi Nathan,

> It looks like you have a commit here that you're confident is correct, then a question that you're less confident about and we should try to investigate together. Am I understanding that correctly? I just want to make sure that I follow up in the right way.

Well, I'm confident that SmallGroup(16,13) is the center product of C4 and Dihedral order 8 is the Pauli Group, but (since filing the ticket) I'm no longer confident that this group really is [GroupNames #16, 13](https://people.maths.bris.ac.uk/~matyd/GroupNames/index.html#c4od4), which is what the PR claims. However if it's not, then the `<link>` on line 5 should change.

I'm just learning group theory (currently on chapter 7 of your book, which is amazing btw), so please forgive my ignorance, but It's not immediately clear to me whether these presentations are the same. One the other hand, I can't immediately rule out that they are not isomorphic.

My understanding is that if the different representations represent the same group, we can find an isomorphism that can map one set of generators into another. Or if they do not, prove that somehow. If we call the Group Explorer generators a, b, c and the GroupNames ones a', b', c', then we would want to find an isomorphism P, where P(a) = a', P(b) = b', P(c) = c'. My understanding is that the identities and inverses would be preserved under any isomorphism. This should give us 1 = P(1) = P(a^4) = a'^4, and so on. However, I haven't been able to find an isomorphism nor prove that one can't exist yet.

Reply:
If the two representations are for the same group, then there will be an isomorphism mapping one representation's generators into the other group, but not necessarily to the set of generators chosen in the second group.  For instance, we can say that the permutation group S_3 is generated by the two elements (1 2 3) and (1 2), or we could say it was generated by the three elements (1 2), (2 3), and (1 3).  There is no isomorphism between those sets of *generators*.

Now, regarding whether Unnamed2-16.group is indeed SmallGroup(16,13):  Yes, it is, as follows.  Let's think through all the 14 groups of order 16, listed in Group Explorer and in GroupNames.  (I'm assuming here that both software packages have the correct set of groups of order 16 to begin with.)

 * I don't think there's any hesitation that D_8, Q_4xZ_2, Z_2xZ_8, D_4xZ_2, Q_8, Z_2xZ_4xZ_2, Z_2xZ_2xZ_2xZ_2, Z_4xZ_2, and Z_16 are all correctly paired up with their corresponding entries in the GroupNames database, so I won't take time on those.
 * We can see that Group Explorer's Modular 16 is correctly paired with GroupNames's M_4(2) because they're the only ones remaining that have 11 subgroups.
 * We can see that Group Explorer's Quasihedral 16 is correctly paired with GroupNames's SD_16 because they're the only ones remaining that have 15 subgroups AND a class equation 1+2+2+2+2+4+4=16.
 * We can see that Group Explorer's Unnamed1_16 is correctly paired with GroupNames's C_4 semidirect-product C_4 because they're the only ones remaining that have 15 subgroups AND a class equation 1+1+1+1+2+2+2+2+2+2=16.
 * This leaves G_4,4 and Unnamed2_16, both of which have 23 subgroups AND a class equation 1+1+1+1+2+2+2+2+2+2=16.  So which one pairs with GroupNames's C_2^2 semidirect-product C_4 and which one pairs with GroupNames's C_4 o D_4?
    * If we check the set of subgroups of Unnamed2_16, we find: one trivial, 7 like Z_2, 4 like Z_4, 3 like Z_2xZ_2, 3 like Z_2xZ_4, 3 like D_4, one like Q_4, and the whole group.  This matches perfectly with the subgroup lattice of C_4 o D_4 in GroupNames.
    * If we check the set of subgroups of G_4,4, we find: one trivial, 7 like Z_2, 4 like Z_4, 7 like Z_2xZ_2, 2 like Z_2xZ_4, one like Z_2xZ_2xZ_2, and the whole group.  This matches perfectly with the subgroup lattice of C_2^2 semidirect-product C_4.

Thus the association set up in Group Explorer is correct.

Reply:
> If the two representations are for the same group, then there will be an isomorphism mapping one representation's generators into the other group, but not necessarily to the set of generators chosen in the second group. For instance, we can say that the permutation group S_3 is generated by the two elements (1 2 3) and (1 2), or we could say it was generated by the three elements (1 2), (2 3), and (1 3). There is no isomorphism between those sets of generators.

Thanks @nathancarter for the thorough explanation and and especially for correcting my understanding about isomorphisms between generators. 

I seems like we can safely resolve this conversation and proceed to rename the Unamed2_16 to Pauli Group (or to C_4oD_4) if you'd like.

We could also rename Unnamed1_16 to C_4 semidirect-product C_4, as you pointed out.

Reply:
Yes I agree with all of that.  I'm not going to merge this right now or do the other change because I'm prepping to teach class this afternoon/evening, but yes--we should do this!  Talk to you again soon.  And thanks again for bringing it up!